app/(api)/auth/user/DELETE.tsx

```diff
@@ -0,0 +1,58 @@
+'use server';
+
+import { NextRequest, NextResponse } from 'next/server';
+import { getDatabase } from '@utils/mongodb/mongoClient.mjs';
+import { HttpError, NotFoundError } from '@utils/response/Errors';
+import { ObjectId } from 'mongodb';
+
+// pass in id of user to delete
+// check if user exists in db before deleting
+/*
+ *   Removes existing user in database
+ *   @param request: {_id: string}
+ *   @returns {ok: boolean, body: {_id: string, email: string, password: string, removedCount: number}, error: string}
+ */
+const collectionName = 'users';
+
+export async function DELETE(request: NextRequest) {
+  try {
+    const db = await getDatabase();
+    const { _id } = await request.json();
+    const userId = ObjectId.createFromHexString(_id);
+
+    // check if user exists in the db
+    const user = await db.collection(collectionName).findOne({ _id: userId });
+
+    // user doesn't exist, so nothing to delete
+    if (!user) {
+      throw new NotFoundError(
+        'User not found, cannot delete non-existent user'
+      );
+    }
+
+    // delete user from db
+    const removedUser = await db
+      .collection(collectionName)
+      .deleteOne({ _id: userId });
+
+    return NextResponse.json(
+      {
+        ok: true,
+        body: {
```

Comment:
Don't think we need to return anything other than a success statement. Maybe look at what people usually return from DELETE API requests in REST APIs.

https://stackoverflow.com/questions/25970523/restful-what-should-a-delete-response-body-contain

maybe just a 204 status is good? If Next.js doesn't like 204 just give 200.

This is all assuming you want to return NextRequest